// @peculiar/asn1-x509-logotype/build/cjs/logotype_audio_info.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogotypeAudioInfo = exports.LogotypeAudioChannels = void 0;
const tslib_1 = require("tslib");
const asn1_schema_1 = require("@peculiar/asn1-schema");
var LogotypeAudioChannels;
(function (LogotypeAudioChannels) {
    LogotypeAudioChannels[LogotypeAudioChannels["mono"] = 1] = "mono";
    LogotypeAudioChannels[LogotypeAudioChannels["stereo"] = 2] = "stereo";
    LogotypeAudioChannels[LogotypeAudioChannels["quad"] = 4] = "quad";
})(LogotypeAudioChannels || (exports.LogotypeAudioChannels = LogotypeAudioChannels = {}));
class LogotypeAudioInfo {
    constructor(params = {}) {
        this.fileSize = 0;
        this.playTime = 0;
        this.channels = LogotypeAudioChannels.mono;
        Object.assign(this, params);
    }
}
exports.LogotypeAudioInfo = LogotypeAudioInfo;
tslib_1.__decorate([
    (0, asn1_schema_1.AsnProp)({ type: asn1_schema_1.AsnPropTypes.Integer })
], LogotypeAudioInfo.prototype, "fileSize", void 0);
tslib_1.__decorate([
    (0, asn1_schema_1.AsnProp)({ type: asn1_schema_1.AsnPropTypes.Integer })
], LogotypeAudioInfo.prototype, "playTime", void 0);
tslib_1.__decorate([
    (0, asn1_schema_1.AsnProp)({ type: asn1_schema_1.AsnPropTypes.Integer })
], LogotypeAudioInfo.prototype, "channels", void 0);
tslib_1.__decorate([
    (0, asn1_schema_1.AsnProp)({ type: asn1_schema_1.AsnPropTypes.Integer, implicit: true, context: 3, optional: true })
], LogotypeAudioInfo.prototype, "sampleRate", void 0);
tslib_1.__decorate([
    (0, asn1_schema_1.AsnProp)({ type: asn1_schema_1.AsnPropTypes.IA5String, implicit: true, context: 4, optional: true })
], LogotypeAudioInfo.prototype, "language", void 0);
