// @peculiar/asn1-x509-logotype/build/cjs/logotype_image_info.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogotypeImageInfo = exports.LogotypeImageType = void 0;
const tslib_1 = require("tslib");
const asn1_schema_1 = require("@peculiar/asn1-schema");
const logotype_image_resolution_1 = require("./logotype_image_resolution");
var LogotypeImageType;
(function (LogotypeImageType) {
    LogotypeImageType[LogotypeImageType["grayScale"] = 0] = "grayScale";
    LogotypeImageType[LogotypeImageType["color"] = 1] = "color";
})(LogotypeImageType || (exports.LogotypeImageType = LogotypeImageType = {}));
class LogotypeImageInfo {
    constructor(params = {}) {
        this.type = LogotypeImageType.color;
        this.fileSize = 0;
        this.xSize = 0;
        this.ySize = 0;
        Object.assign(this, params);
    }
}
exports.LogotypeImageInfo = LogotypeImageInfo;
tslib_1.__decorate([
    (0, asn1_schema_1.AsnProp)({ type: asn1_schema_1.AsnPropTypes.Integer, context: 0, implicit: true, defaultValue: LogotypeImageType.color })
], LogotypeImageInfo.prototype, "type", void 0);
tslib_1.__decorate([
    (0, asn1_schema_1.AsnProp)({ type: asn1_schema_1.AsnPropTypes.Integer })
], LogotypeImageInfo.prototype, "fileSize", void 0);
tslib_1.__decorate([
    (0, asn1_schema_1.AsnProp)({ type: asn1_schema_1.AsnPropTypes.Integer })
], LogotypeImageInfo.prototype, "xSize", void 0);
tslib_1.__decorate([
    (0, asn1_schema_1.AsnProp)({ type: asn1_schema_1.AsnPropTypes.Integer })
], LogotypeImageInfo.prototype, "ySize", void 0);
tslib_1.__decorate([
    (0, asn1_schema_1.AsnProp)({ type: logotype_image_resolution_1.LogotypeImageResolution })
], LogotypeImageInfo.prototype, "resolution", void 0);
tslib_1.__decorate([
    (0, asn1_schema_1.AsnProp)({ type: asn1_schema_1.AsnPropTypes.IA5String, context: 4, implicit: true, optional: true })
], LogotypeImageInfo.prototype, "language", void 0);
